src/synthetix/StakingRewards.sol

```diff
@@ -162,11 +162,14 @@ contract StakingRewards is IStakingRewards, Pausable, ReentrancyGuard {
         emit Recovered(tokenAddress, tokenAmount);
     }
 
-    function setRewardsDuration(uint256 _rewardsDuration) external onlyOwner {
-        require(
-            block.timestamp > periodFinish,
-            "Previous rewards period must be complete before changing the duration for the new period"
-        );
+    function setRewardsDuration(uint256 _rewardsDuration) external onlyOwner updateReward(address(0)) {
```

Comment:
Seeing that the `interfaces` dir has an `IstakingRewards.sol` file, which also include the authed functions but does not include `setRewardsDuration` and `recoverERC20`. This might be a good time to add these.

BTW the Synthetix interface [file](https://github.com/Synthetixio/synthetix/blob/5e9096ac4aea6c4249828f1e8b95e3fb9be231f8/contracts/interfaces/IStakingRewards.sol) seems to only include the non-authed functions, but if the authed files are added then all should be include.

Reply:
Done